Assets/MixedRealityToolkit.Examples/Experimental/Audio/Scripts/MicrophoneAmplitudeDemo.cs

```diff
@@ -0,0 +1,208 @@
+
+        private void Awake()
+        {
+            // We do not wish to play the ambient room sound from the audio source.
+            gameObject.GetComponent<AudioSource>().volume = 0.0f;
+
+            if (VisibleMaterial != null)
+            {
+#if UNITY_EDITOR
+                // Cache the initial material settings.
+                defaultMaterialColor = VisibleMaterial.GetColor("_WireColor");
+                defaultWireThickness = VisibleMaterial.GetInt("_WireThickness");
+#endif // UNITY_EDITOR
+
+                VisibleMaterial.SetColor("_WireColor", Color.blue);
+            }
+
+            micStream = new WindowsMicrophoneStream();
+            micStream.Gain = 1.0f;
+
+            // Initialize the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.Initialize(WindowsMicrophoneStreamType.HighQualityVoice);
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to initialize the microphone stream. {result}");
+            }
+
+            // Start the microphone stream.
+            // Do not keep the data and do not preview.
+            result = micStream.StartStream(false, false);
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to start the microphone stream. {result}");
+            }
+        }
+
+        private void OnDestroy()
+        {
+            // Stop the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.StopStream();
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to stop the microphone stream. {result}");
+            }
+
+            // Uninitialize the microphone stream.
+            micStream.Uninitialize();
+            micStream = null;
+
+#if UNITY_EDITOR
+            // Restore the initial material settings.
+            if (VisibleMaterial != null)
+            {
+                VisibleMaterial.SetColor("_WireColor", defaultMaterialColor);
+                VisibleMaterial.SetInt("_WireThickness", defaultWireThickness);
+            }
+#endif // UNITY_EDITOR
+        }
+
+        private void OnDisable()
+        {
+            // Pause the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.Pause();
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to pause the microphone stream. {result}");
+            }
+        }
+
+        private void OnEnable()
+        {
+            // Resume the microphone stream.
+            WindowsMicrophoneStreamErrorCode result = micStream.Resume();
+            if (result != WindowsMicrophoneStreamErrorCode.Success)
+            {
+                Debug.Log($"Failed to resume the microphone stream. {result}");
+            }
+        }
+
+        private static int maxWireThickness = 750;
```

Comment:
Nit: move this to the other member variables out of the function section and add some doccomments